database/mediabackfillrequest.go

```diff
@@ -0,0 +1,126 @@
+// mautrix-whatsapp - A Matrix-WhatsApp puppeting bridge.
+// Copyright (C) 2022 Tulir Asokan, Sumner Evans
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package database
+
+import (
+	"database/sql"
+	"errors"
+
+	_ "github.com/mattn/go-sqlite3"
+	log "maunium.net/go/maulogger/v2"
+	"maunium.net/go/mautrix/id"
+)
+
+type MediaBackfillRequestStatus int
+
+const (
+	MediaBackfillRequestStatusNotRequested MediaBackfillRequestStatus = iota
+	MediaBackfillRequestStatusRequested
+	MediaBackfillRequestStatusRequestFailed
+)
+
+type MediaBackfillRequestQuery struct {
+	db  *Database
+	log log.Logger
+}
+
+type MediaBackfillRequest struct {
+	db  *Database
+	log log.Logger
+
+	UserID    id.UserID
+	PortalKey *PortalKey
+	EventID   id.EventID
+	Status    MediaBackfillRequestStatus
+	Error     string
```

Comment:
It might be nicer to store the `MediaKey` in here too to avoid having to fetch all the events from Matrix when going through the queue.

(just need to remember to set it to null after attempting to request the media)

Reply:
`portal.requestMediaRetry` seems to end up needing more than just the media key to send the retry receipt which is why I stored the event ID

Why would you set it to null? Is that an alternative to storing the status?